feat: disable replay in decide if over quota

frankh opened the pull request:
## Problem

When session replay is over the billing limit the client JS continues sending requests which we drop silently.

We recently added a change to change the capture response to include that it was over quota and to stop subsequent sends, but we still send the first one.

Include it in the decide check to prevent this (this also means we can be more flexible with our capture response and not necessarily need to include the quota limit there)

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

<!-- Yes / no / it doesn't have an impact. -->

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

frankh committed:
disable replay in decide if over quota

benjackwhite commented on posthog/api/decide.py:
```diff
@@ -291,13 +291,23 @@ def get_decide(request: HttpRequest):
         if team.session_recording_opt_in and (
             on_permitted_recording_domain(team, request) or not team.recording_domains
         ):
+            if settings.EE_AVAILABLE:
```
I like the idea but can we add another flag for this. We haven't tested that function in decide and any number of things could go wrong so would nice to be able to revert easily here.

benjackwhite replied:
Also a test would be really necessary here

frankh replied:
yeah i wanted feedback on whether it was a complete non-starter for some reason i hadn't thought of before adding tests

Phanatic replied:
Instead of depending on a revert, I would have suggested a feature flag here. But that would end up creating an ouroboros. The irony of using feature flags inside the feature flag endpoint :)

Phanatic reviewed (commented):
Looks good to me.

frankh committed:
add test

frankh committed:
fix token type

frankh committed:
fix test

benjackwhite reviewed (changes requested):
Given that this can apply for events as well maybe we want to do a more straightforward object like we did for capture. That way we can add events, or any other thing easily later and it's really clear that the quota limiting is in place (and not just that replay is disabled). 

Could be in addition to disabling it

frankh committed:
return quotaLimited object in decide if quota limited

this allows the client to know why recordings (or potentially other
services) are disabled

benjackwhite commented on posthog/api/decide.py:
```diff
@@ -245,7 +246,18 @@ def get_decide(request: HttpRequest):
+                )
+
+                if token in limited_tokens_recordings:
+                    response["quotaLimited"] = ["sessionRecording"]
```
```suggestion
                    response["quotaLimited"] = ["recordings"]
```

benjackwhite committed:
Fixes

benjackwhite reviewed (approved):
Pushed some changes as it was easier than describing. The quota names should be the same as in capture and we should always respond with it if set (even if recordings is disabled) as it will be easier to reason on and debug

benjackwhite committed:
fix
